File: Code/27.remove-element.cpp
/*
 * @lc app=leetcode id=27 lang=cpp
 *
 * [27] Remove Element
 */
#include <vector>
using namespace std;
// @lc code=start
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        if (nums.empty())
        {
            return 0;
        }
        int slow = 0;
        for(int fast = 0; fast<nums.size(); fast++){
            if(nums[fast]  != val){
                nums[slow] = nums[fast];
                slow++;
            }
        }
        return slow;
        
    }
};
// @lc code=end

File: 27.remove-element.py
#
# @lc app=leetcode id=27 lang=python3
#
# [27] Remove Element
#

# @lc code=start
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        k=0
        right = len(nums)-1
        left = 0
        while left<=right:
            if nums[left] == val:
                nums[left]=nums[right]
                right-=1
                
            else:
                left+=1
                k+=1
        return k
# @lc code=end

